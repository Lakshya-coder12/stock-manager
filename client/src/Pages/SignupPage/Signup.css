.redirect {
  margin-top: 15px;
  color: #1976c2;
  font-weight: 600;
  text-align: center;
}

.redirect:hover {
  text-decoration: underline;
  cursor: pointer;
}
