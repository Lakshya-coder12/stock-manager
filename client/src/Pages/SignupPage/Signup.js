import React, { Component } from "react";
import Button from "../../Components/Button/Button";
import "./Signup.css";
import PropTypes from "prop-types";

export class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      email: "",
      password: "",
      confirmPassword: "",
    };
  }
  handleInputChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  };

  handleRedirect = () => {
    this.props.history.push("/login");
  };

  handleSignup = (e) => {
    e.preventDefault();
    const formData = {
      name: this.state.name,
      email: this.state.email,
      password: this.state.password,
      confirmPassword: this.state.confirmPassword,
    };
    this.props.signup(formData, this.props.history);
  };
  render() {
    const { name, email, password, confirmPassword } = this.state;
    return (
      <>
        <form className="form" onSubmit={this.handleSignup}>
          <div className="input-element">
            <label className="label">Name</label>
            <input
              required
              name="name"
              onChange={this.handleInputChange}
              value={name}
              className="input"
              type="text"
            />
          </div>
          <div className="input-element">
            <label className="label">Email</label>
            <input
              required
              name="email"
              onChange={this.handleInputChange}
              value={email}
              className="input"
              type="email"
            />
          </div>
          <div className="input-element">
            <label className="label">Password</label>
            <input
              required
              name="password"
              className="input"
              onChange={this.handleInputChange}
              value={password}
              type="password"
            />
          </div>
          <div className="input-element">
            <label className="label">Confirm Password</label>
            <input
              required
              name="confirmPassword"
              onChange={this.handleInputChange}
              value={confirmPassword}
              className="input"
              type="password"
            />
          </div>
          <div className="form-btn">
            <Button>Sign Up</Button>
          </div>
          <div onClick={this.handleRedirect} className="redirect">
            Already have an account? Click here to sign in
          </div>
        </form>
      </>
    );
  }
}

Signup.propTypes = {
  signup: PropTypes.func.isRequired,
};

export default Signup;
