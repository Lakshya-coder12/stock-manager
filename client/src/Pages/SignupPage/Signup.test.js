import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Signup from "./Signup";

configure({ adapter: new Adapter() });

describe("Testing Signup page", () => {
  it("matches snapshot", () => {
    const signup = jest.fn();
    const wrapper = shallow(<Signup signup={signup} />);
    expect(wrapper).toMatchSnapshot();
  });
});
