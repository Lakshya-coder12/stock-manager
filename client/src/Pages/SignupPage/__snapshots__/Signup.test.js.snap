// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Signup page matches snapshot 1`] = `
<Fragment>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="input-element"
    >
      <label
        className="label"
      >
        Name
      </label>
      <input
        className="input"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="input-element"
    >
      <label
        className="label"
      >
        Email
      </label>
      <input
        className="input"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="input-element"
    >
      <label
        className="label"
      >
        Password
      </label>
      <input
        className="input"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="input-element"
    >
      <label
        className="label"
      >
        Confirm Password
      </label>
      <input
        className="input"
        name="confirmPassword"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-btn"
    >
      <Button>
        Sign Up
      </Button>
    </div>
    <div
      className="redirect"
      onClick={[Function]}
    >
      Already have an account? Click here to sign in
    </div>
  </form>
</Fragment>
`;
